crates/wasm-compose/src/encoding.rs

```diff
@@ -143,8 +143,11 @@ impl<'a> TypeEncoder<'a> {
         }
 
         for (name, url, ty) in exports {
-            let ty = self.component_entity_type(&mut encoded, &mut types, ty);
-            encoded.export(name, url, ty);
+            let type_definiton = self.0.type_from_id(ty.id().unwrap()).unwrap();
+            let export = self.component_entity_type(&mut encoded, &mut types, ty);
+            // Point the type index to the export instead of the concrete type definition
+            types.insert(PtrKey(type_definiton), encoded.type_count());
+            encoded.export(name, url, export);
```

Comment:
The test failure is because we only want to do this map update when the item being exported is a type, so that the type map points at the exported (i.e. "named") index rather than the original type definition index.

To do that, I think something like this would work (here and the other places we needed this change):

```suggestion
            let id = match &ty {
                ComponentEntityType::Type { created: id, .. } => Some(*id),
                _ => None,
            };

            let ty = self.component_entity_type(&mut encoded, &mut types, ty);
            if let Some(id) = id {
                let ty = self.0.type_from_id(id).unwrap();
                types.insert(PtrKey(ty), encoded.type_count());
            }

            encoded.export(name, url, ty);
```